File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/plugin/fetchoptions/PluginFetchOptions.js
define([ "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/person/fetchoptions/PersonFetchOptions", 
         "as/dto/common/fetchoptions/EmptyFetchOptions", "as/dto/plugin/fetchoptions/PluginSortOptions" ], function(
		stjs, FetchOptions) {
	var PluginFetchOptions = function() {
	};
	stjs.extend(PluginFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.plugin.fetchoptions.PluginFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.registrator = null;
		prototype.script = null;
		prototype.sort = null;

		prototype.withRegistrator = function() {
			if (this.registrator == null) {
				var PersonFetchOptions = require("as/dto/person/fetchoptions/PersonFetchOptions");
				this.registrator = new PersonFetchOptions();
			}
			return this.registrator;
		};
		prototype.withRegistratorUsing = function(fetchOptions) {
			return this.registrator = fetchOptions;
		};
		prototype.hasRegistrator = function() {
			return this.registrator != null;
		};
		prototype.withScript = function() {
			if (this.script == null) {
				var EmptyFetchOptions = require("as/dto/common/fetchoptions/EmptyFetchOptions");
				this.script = new EmptyFetchOptions();
			}
			return this.script;
		};
		prototype.withScriptUsing = function(fetchOptions) {
			return this.script = fetchOptions;
		}
		prototype.hasScript = function() {
			return this.script != null;
		};
		prototype.sortBy = function() {
			if (this.sort == null) {
				var PluginSortOptions = require("as/dto/plugin/fetchoptions/PluginSortOptions");
				this.sort = new PluginSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		registrator : "PersonFetchOptions",
		script : "EmptyFetchOptions",
		sort : "PluginSortOptions"
	});
	return PluginFetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/plugin/fetchoptions/PluginFetchOptions.java
/*
 * Copyright 2018 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.fetchoptions;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.fetchoptions.PersonFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.Plugin;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 *
 */
@JsonObject("as.dto.plugin.fetchoptions.PluginFetchOptions")
public class PluginFetchOptions extends FetchOptions<Plugin> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private PersonFetchOptions registrator;

    @JsonProperty
    private PluginSortOptions sort;
    
    @JsonProperty
    private boolean script;

    public PersonFetchOptions withRegistrator()
    {
        if (registrator == null)
        {
            registrator = new PersonFetchOptions();
        }
        return registrator;
    }

    public PersonFetchOptions withRegistratorUsing(PersonFetchOptions fetchOptions)
    {
        return registrator = fetchOptions;
    }

    public boolean hasRegistrator()
    {
        return registrator != null;
    }
    @Override
    public PluginSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new PluginSortOptions();
        }
        return sort;
    }
    
    @JsonIgnore
    public PluginFetchOptions withScript()
    {
        script = true;
        return this;
    }
    
    @JsonIgnore
    public boolean isWithScript()
    {
        return script;
    }

    @Override
    public PluginSortOptions getSortBy()
    {
        return sort;
    }

    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("Plugin", this);
        f.addFetchOption("Registrator", registrator);
        return f;
    }

}
